mhkit/wave/resource.py

```diff
@@ -684,3 +686,259 @@ def func(kk):
     k = pd.DataFrame(k0, index=f, columns=['k'])
     
     return k
+
+def environmental_contour(x1, x2, dt, period, PCA=None, size_bin=250, 
+                          nb_steps=1000, return_PCA=False):
+    '''    
+    Calculates environmental contours of extreme sea 
+    states using the improved joint probability distributions 
+	with the inverse first-order reliability method (IFORM) 
+    probability for the desired return period (period). Given the 
+    period of interest a circle of iso-probability is created in the
+    in the PCA joint probability (x1, x2) reference frame.  
+    Using the joint probability value the CDF of the marginal 
+    distribution is used to find the quantile of each component. 
+    Finally, using the improved PCA methodology
+    the component 2 contour lines are calculated from component 1 using 
+    the relationships defined in _principal_component_analysis.	
+
+    Eckert-Gallup, A. C., Sallaberry, C. J., Dallman, A. R., & 
+    Neary, V. S. (2016). Application of principal component 
+    analysis (PCA) and improved joint probability distributions to 
+    the inverse first-order reliability method (I-FORM) for predicting 
+    extreme sea states. Ocean Engineering, 112, 307-319.
+
+    Parameters
+    ----------
+    x1: array like
+        Component 1 data
+    x2: array like
+        Component 2 data        	
+    dt : float
+        x1 and x2 sample rate (seconds) 
+    period : np.array
+        Desired return period (years) for calculation of environmental
+        contour, can be a scalar or a vector. If vector, must be an n by 1 array. 
+    PCA: dict (optional)
```

Comment:
Sterling, I do not fully understand the scope of the PCA variable. Is this in case the user wants to re-use the PCA from a previous time running this code? Or can they compute the output from PCA using their own code/ methods? Please make this doctoring more descriptive. Also, we should not reference a hidden function in the doctoring since the hidden function will not appear in the MHKiT documentation.

Reply:
Passing the PCA variable allows you to skip calculating a PCA for the passed x1, x2. In the next phase of this PR we will be sampling the environmental contours function this function will use the same PCA each time.

Reply:
I know this is an outdated comment, but I shared @rpauly18's confusion with providing a PCA and made minor updates to the PCA description based on this discussion.